more_itertools/tests/test_more.py

```diff
@@ -2311,3 +2311,26 @@ def test_iterable_substitutes(self):
         actual = list(mi.replace(iterable, pred, substitutes))
         expected = ['_', '_', 1, '_', '_', 3, '_', '_']
         self.assertEqual(actual, expected)
+
+
+class PartitionsTest(TestCase):
+    def test_basic(self):
+        iterable = 'abcd'
+        actual = list(mi.partitions(iterable))
+        expected = [
+            ['abcd'],
+            ['a', 'bcd'],
+            ['ab', 'cd'],
+            ['abc', 'd'],
+            ['a', 'b', 'cd'],
+            ['a', 'bc', 'd'],
+            ['ab', 'c', 'd'],
+            ['a', 'b', 'c', 'd']
+        ]
+        self.assertEqual(actual, expected)
+
+    def test_empty(self):
+        iterable = []
+        actual = list(mi.partitions(iterable))
+        expected = []
```

Comment:
I'm not sure now that this is correct.

>The empty set ∅ has exactly one partition, namely ∅ (https://en.wikipedia.org/wiki/Partition_of_a_set#Examples)

So, now I think it should be `[[[]]]`. Also, this is easier to implement and makes things more consistent.

Reply:
I've changed the test and the code.